packages/web-fragments/src/gateway/fragment-gateway.ts

```diff
@@ -46,9 +46,28 @@ export class FragmentGateway {
 			);
 		}
 
+		// TODO: defensively deep clone the config object, but preserve any functions or property getters
+
 		// default to true
 		fragmentConfig.piercing ??= true;
 
+		if (fragmentConfig.iframeHeaders) {
+			// normalize header name casing to Camel-Case
+			fragmentConfig.iframeHeaders = Object.fromEntries(
+				Object.entries(fragmentConfig.iframeHeaders!).map(([key, value]) => [
```

Comment:
We are inside of the if clause so this is truthy, no need to assert non-null

Reply:
done